app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -127,6 +133,16 @@ public void onReceive(Context context, Intent intent) {
                 showMessage(msg);
             }
         };
+
+        openDownloadReceiver = new BroadcastReceiver() {
```

Comment:
Avoid sharing the receiver: it's still a security problem that would accept outside broadcasts.

Reply:
Sharing?

Reply:
Ah this is invalid, sorry.